### PR TITLE
Add archive options to planet-orders-request

### LINKED ISSUE #970: The --cloudconfig CLI option  has an innaccurate name
The `--cloudconfig` option of the `planet orders create` CLI command has an inaccurate name, as it is not limited to cloud delivery configurations. 

It is entirely possible to provide a local delivery config using the `--cloudconfig` option, e.g.:
```
{  
    "archive_type": "zip",
    "single_archive": true,
    "archive_filename": "{{name}}_{{order_id}}.zip"
}
```

The option should be named `--delivery` or `--deliveryconfig`, and [the docs](https://planet-sdk-for-python-v2.readthedocs.io/en/latest/cli/cli-orders/#cloud-delivery) updated accordingly. The [equivalent SDK method](https://planet-sdk-for-python-v2.readthedocs.io/en/latest/python/sdk-reference/#planet.order_request.delivery) is more accurately named `delivery`

### PR DESCRIPTION
**Pull request by sgillies**
And make --delivery and alias for --cloudconfig.

Resolves #970

### CODE REVIEW

File: planet/cli/orders.py [hunk 1]
```diff
@@ -253,11 +253,26 @@ async def create(ctx, request: str, pretty):
...
+              show_default=True,
+              help="Templated filename for archived bundles or orders.")
+@click.option('--single-archive',
+              is_flag=True,
+              default=False,
+              show_default=True,
+              help="Optionally zip archive all item bundles together.")
 @click.option(
     '--cloudconfig',
+    '--delivery',
```
Comment (sgillies) [inline planet/cli/orders.py:270]: Here's the alias.

File: planet/cli/orders.py [hunk 2]
```diff
@@ -253,11 +253,26 @@ async def create(ctx, request: str, pretty):
...
     type=types.JSON(),
-    help="""Credentials for cloud storage provider to enable cloud delivery of
-    data. Can be a json string, filename, or '-' for stdin.""")
+    help=("Credentials for cloud storage provider to enable cloud delivery of "
+          "data. Can be a json string, filename, or '-' for stdin. "
+          "Optionally, archive parameters can be set using the same JSON. "
+          "The --delivery option is an alias for this use case."))
```
Comment (sgillies) [inline planet/cli/orders.py:275]: Too confusing? Or okay?

File: planet/cli/orders.py [hunk 3]
```diff
@@ -303,14 +321,12 @@ async def request(ctx,
         except planet.exceptions.ClientError as e:
             raise click.BadParameter(e)
 
-    if cloudconfig:
-        delivery = planet.order_request.delivery(cloud_config=cloudconfig)
-        if "google_earth_engine" in cloudconfig:
```
Comment (sgillies) [inline planet/cli/orders.py:308]: I moved this test closer to where it is needed.

File: planet/order_request.py [hunk 4]
```diff
@@ -199,13 +199,17 @@ def delivery(archive_type: Optional[str] = None,
     if archive_type:
         archive_type = specs.validate_archive_type(archive_type)
 
-    fields = ['archive_type', 'single_archive', 'archive_filename']
-    values = [archive_type, single_archive, archive_filename]
+    config: Dict[str, Any] = {}
 
-    config = dict((k, v) for k, v in zip(fields, values) if v)
+    if archive_type:
```
Comment (sgillies) [inline planet/order_request.py:204]: This covers some of the needs of #963, I think.

File: planet/order_request.py [hunk 5]
```diff
@@ -199,13 +199,17 @@ def delivery(archive_type: Optional[str] = None,
...
+        config.update(archive_type=archive_type,
+                      archive_filename=archive_filename
+                      or '{{name}}-{{order_id}}.zip',
```
Comment (sgillies) [inline planet/order_request.py:207]: Here's a default archive filename.

Comment (sgillies) [reply planet/order_request.py:207]: Note that unless the archive type is given, there are no archive items in the "delivery" config.

**Comment by sgillies** [comment]
@kevinlacaille @tbarsballe what do you think of this attempt to solve #970 by adding the missing individual archive options that complement `--cloudconfig` as well as a `--delivery` alias?

File: planet/cli/orders.py (hunk 2 above)
Comment (tbarsballe) [reply planet/cli/orders.py:275]: If feels a little unclear as is. I would reword such that --cloudconfig is the alias, and merge the archive parameter line into the first sentence. E.g.:
```
"Delivery configuration, which may include credentials for a cloud storage "
"provider, to enable cloud delivery of data, and/or parameters for bundling "
"deliveries as zip archives. Can be a json string, filename, or '-' for stdin. "
"The --cloudconfig option is an alias for this use case."
```

File: planet/cli/orders.py [hunk 6]
```diff
@@ -253,11 +253,26 @@ async def create(ctx, request: str, pretty):
...
+              help="Optionally zip archive each item bundle.")
+@click.option('--archive-filename',
+              default='{{name}}-{{order_id}}.zip',
+              show_default=True,
+              help="Templated filename for archived bundles or orders.")
+@click.option('--single-archive',
+              is_flag=True,
+              default=False,
+              show_default=True,
+              help="Optionally zip archive all item bundles together.")
```
Comment (tbarsballe) [inline planet/cli/orders.py:267]: This is a great addition, thanks for including it.

**Review by tbarsballe — COMMENTED** [review]
Looks pretty good to me. I like keeping --cloudconfig an having --delivery as an alias - that obviate's my concern around changing the parameter being too user-unfriendly.

I did suggest a rewording of the parameter help, though it is alright as is.

File: planet/cli/orders.py (hunk 6 above)
Comment (sgillies) [reply planet/cli/orders.py:267]: Yes indeed, this was an oversight. It's easy to imagine users only using, for example, `--archive-type zip` and no cloud config at all.

**Comment by sgillies** [comment]
Thanks for the feedback @tbarsballe !

File: planet/cli/orders.py (hunk 1 above)
Comment (tbarsballe) [reply planet/cli/orders.py:269]: Great, looks good to me.

**Review by tbarsballe — APPROVED** [review]
